include/xgboost/parameter.h

```diff
@@ -78,4 +79,28 @@ class FieldEntry<EnumClass> : public FieldEntry<int> {  \
 }  /* namespace parameter */  \
 }  /* namespace dmlc */
 
-#endif  // XGBOOST_ENUM_CLASS_PARAM_H_
+namespace xgboost {
+template <typename Type>
+struct XGBoostParameter : public dmlc::Parameter<Type> {
+  bool initialised_ {false};
+
+ public:
+  bool& Initialised() { return initialised_; }
```

Comment:
This looks a bit unexpected
1. Do we want to permit to change initialized_ flag from outside?
2. If so shouldn't we create a separate method for that?
Also note that one cannot read this flag if the instance of XGBoostParameter is constant

Reply:
@trams Will fix it.  This combines an new commit in `dmlc-core`.  Just got it merged today.  Will clean them up.